src/CommunityToolkit.Maui.Camera/CameraManager.macios.cs

```diff
@@ -0,0 +1,299 @@
+		this.flashMode = flashMode.ToPlatform();
+	}
+
+	public partial void UpdateZoom(float zoomLevel)
+	{
+		if (!IsInitialized || captureDevice is null)
+		{
+			return;
+		}
+
+		if (zoomLevel < (float)captureDevice.MinAvailableVideoZoomFactor || zoomLevel > (float)captureDevice.MaxAvailableVideoZoomFactor)
+		{
+			return;
+		}
+
+		captureDevice.LockForConfiguration(out NSError error);
+		if (error is not null)
+		{
+			Console.WriteLine(error);
+			Debug.WriteLine(error);
+			return;
+		}
+
+		captureDevice.VideoZoomFactor = zoomLevel;
+		captureDevice.UnlockForConfiguration();
+	}
+
+	public async partial ValueTask UpdateCaptureResolution(Size resolution, CancellationToken token)
+	{
+		if (captureDevice is null)
+		{
+			return;
+		}
+
+		captureDevice.LockForConfiguration(out NSError error);
+		if (error is not null)
+		{
+			Trace.WriteLine(error);
+			return;
+		}
+
+		if (cameraView.SelectedCamera is null)
+		{
+			await cameraProvider.RefreshAvailableCameras(token);
+			cameraView.SelectedCamera = cameraProvider.AvailableCameras?.FirstOrDefault() ?? throw new CameraViewException("No camera available on device");
+		}
+
+		var filteredFormatList = cameraView.SelectedCamera.SupportedFormats.Where(f =>
+		{
+			var d = ((CMVideoFormatDescription)f.FormatDescription).Dimensions;
+			return d.Width <= resolution.Width && d.Height <= resolution.Height;
+		});
+
+		filteredFormatList = (filteredFormatList.Any() ? filteredFormatList : cameraView.SelectedCamera.SupportedFormats)
+			.OrderByDescending(f =>
+			{
+				var d = ((CMVideoFormatDescription)f.FormatDescription).Dimensions;
+				return d.Width * d.Height;
+			});
+
+		if (filteredFormatList.Any())
+		{
+			captureDevice.ActiveFormat = filteredFormatList.First();
+		}
+
+		captureDevice.UnlockForConfiguration();
+	}
+
+	protected virtual async partial Task PlatformConnectCamera(CancellationToken token)
+	{
+		if (cameraProvider.AvailableCameras is null)
+		{
+			await cameraProvider.RefreshAvailableCameras(token);
+
+			if (cameraProvider.AvailableCameras is null || cameraProvider.AvailableCameras.Count < 1)
+			{
+				throw new CameraViewException("No camera available on device");
+			}
+		}
+
+		await PlatformStartCameraPreview(token);
+	}
+
+	protected virtual async partial Task PlatformStartCameraPreview(CancellationToken token)
+	{
+		if (captureSession is null)
+		{
+			return;
+		}
+
+		captureSession.BeginConfiguration();
+
+		foreach (var input in captureSession.Inputs)
+		{
+			captureSession.RemoveInput(input);
+			input.Dispose();
+		}
+
+		if (cameraView.SelectedCamera is null)
+		{
+			await cameraProvider.RefreshAvailableCameras(token);
+			cameraView.SelectedCamera = cameraProvider.AvailableCameras?.FirstOrDefault() ?? throw new CameraViewException("No camera available on device");
+		}
+
+		captureDevice = cameraView.SelectedCamera.CaptureDevice ?? throw new CameraViewException($"No Camera found");
+		captureInput = new AVCaptureDeviceInput(captureDevice, out var err);
```

Comment:
we declare the `err` but never use it, is that intentional?

Reply:
Changed to `_` ✅